test/k8sT/Policies.go

```diff
@@ -1316,6 +1319,312 @@ var _ = SkipDescribeIf(func() bool {
+			appPods          map[string]string
+			namespaceForTest string
+		)
+
+		BeforeAll(func() {
+			namespaceForTest = helpers.GenerateNamespaceForTest("")
+			kubectl.NamespaceDelete(namespaceForTest)
+			kubectl.NamespaceCreate(namespaceForTest).ExpectSuccess("could not create namespace")
+			kubectl.Apply(helpers.ApplyOptions{FilePath: demoPath, Namespace: namespaceForTest}).ExpectSuccess("could not create resource")
+
+			err := kubectl.WaitforPods(namespaceForTest, "-l zgroup=testapp", helpers.HelperTimeout)
+			Expect(err).Should(BeNil(), "Test pods are not ready after timeout")
+
+			ciliumPod, err = kubectl.GetCiliumPodOnNode(helpers.K8s1)
+			Expect(err).Should(BeNil(), "cannot get CiliumPod")
+
+			clusterIP, _, err = kubectl.GetServiceHostPort(namespaceForTest, app1Service)
+			Expect(err).To(BeNil(), "Cannot get service in %q namespace", namespaceForTest)
+			appPods = helpers.GetAppPods(apps, namespaceForTest, kubectl, "id")
+			logger.WithFields(logrus.Fields{
+				"ciliumPod": ciliumPod,
+				"clusterIP": clusterIP}).Info("Initial data")
+
+		})
+
+		AfterAll(func() {
+			kubectl.NamespaceDelete(namespaceForTest)
+			kubectl.Delete(demoPath)
+		})
+
+		BeforeEach(func() {
+			kubectl.CiliumExecMustSucceed(context.TODO(),
+				ciliumPod, fmt.Sprintf("cilium config %s=%s",
+					helpers.PolicyEnforcement, helpers.PolicyEnforcementDefault))
+
+			err := kubectl.CiliumEndpointWaitReady()
+			Expect(err).To(BeNil(), "Endpoints are not ready after timeout")
+
+			err = kubectl.WaitforPods(namespaceForTest, "-l zgroup=testapp", helpers.HelperTimeout)
+			Expect(err).Should(BeNil())
+
+		})
+
+		AfterEach(func() {
+			cmd := fmt.Sprintf("%s delete --all cnp,ccnp,netpol -n %s", helpers.KubectlCmd, namespaceForTest)
+			_ = kubectl.Exec(cmd)
+		})
+
+		policyMetricsTest := func(expectsSuccess bool) {
+			policyCount := 0
+			cmd := "cilium metrics list -o json | jq '.[] | select( .name == \"cilium_policy\" ).value'"
+			res := kubectl.CiliumExecContext(context.Background(), ciliumPod, cmd)
+			for _, cnt := range res.ByLines() {
+				policyCount, _ = strconv.Atoi(cnt)
+			}
+
+			policyUsedCount := 0
+			cmd = "cilium metrics list -o json | jq '.[] | select( .name == \"cilium_policy_used\" ).value'"
+			res = kubectl.CiliumExecContext(context.Background(), ciliumPod, cmd)
+			for _, cnt := range res.ByLines() {
+				policyUsedCount, _ = strconv.Atoi(cnt)
+			}
+
+			policyUnusedCount := 0
+			cmd = "cilium metrics list -o json | jq '.[] | select( .name == \"cilium_policy_unused\" ).value'"
+			res = kubectl.CiliumExecContext(context.Background(), ciliumPod, cmd)
+			for _, cnt := range res.ByLines() {
+				policyUnusedCount, _ = strconv.Atoi(cnt)
+			}
+
+			totalPolicyCount := policyUsedCount + policyUnusedCount
+
+			if expectsSuccess {
+				Expect(policyCount == totalPolicyCount).Should(Equal(true), "Policy count does not match with PolicyUsed and PolicyUnUsed\n")
+			}
+		}
+
+		It("checks all policies metrics", func() {
```

Comment:
Policy metrics would seem like a good candidate for unit testing rather than integration testing. With unit testing we'd get much faster test iteration and could also cover all the changed code paths much easier by mocking rather than creating real deployments.